### PR TITLE
feat(transformer) Adds glossary terms transformer

### PR DESCRIPTION
**Pull request by ecooklin**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,6 +120,7 @@ Here are the companies that have officially adopted DataHub. Please feel free to
 - [LinkedIn](http://linkedin.com)
 - [Peloton](https://www.onepeloton.com)
 - [Saxo Bank](https://www.home.saxo)
+- [Stash](https://www.stash.com)
```
Comment (gabe-lyons) [inline README.md:123]: 🥳

File: metadata-ingestion/transformers.md [hunk 2]
```diff
@@ -25,6 +25,24 @@ transformers:
...
+The config, which we’d append to our ingestion recipe YAML, would look like this:
+
+```yaml
+transformers:
+  - type: "pattern_add_dataset_tags"
+    config:
+      tag_pattern:
+        rules:
+          ".*example1.*": ["urn:li:tag:NeedsDocumentation", "urn:li:tag:Legacy"]
+          ".*example2.*": ["urn:li:tag:NeedsDocumentation"]
```
Comment (gabe-lyons) [inline metadata-ingestion/transformers.md:41]: thanks for adding this extension!

**Review by gabe-lyons — COMMENTED** [review]
this looks good on my end but I will defer to someone with more python experience to approve 👍

**Review by shirshanka — APPROVED** [review]
LGTM! Thanks for this contribution @ecooklin 🙏

**Pull request merged**